File: Boj-Kotlin&Java/백준2435번기상청인턴신현수.java
package 백준;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class 백준2435번기상청인턴신현수 {
    static int n,k;
    static int[] arr;
    static int max=Integer.MIN_VALUE;
    static int sum=0;
    public static void main(String[] args) throws IOException {
        BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
        String[] nk=br.readLine().split(" ");
        n=Integer.parseInt(nk[0]);
        k=Integer.parseInt(nk[1]);
        arr=new int[n];
        String[] input=br.readLine().split(" ");
        for(int i=0; i<input.length; i++)
            arr[i]=Integer.parseInt(input[i]);
        for(int i=0; i<=n-k; i++){
            for(int j=i; j<i+k; j++)
                sum+=arr[j];
            max=Math.max(max,sum);
            sum=0;
        }
        System.out.println(max);
    }
}

File: Boj-Kotlin&Java/백준 2435번 기상청 인턴 신현수.kt
package 백준

import java.util.*
import kotlin.math.max

//입력형식에 문제가있는 문제라 kotlin 평소방식불가능
fun main() {
    val sc=Scanner(System.`in`)
    val n=sc.nextInt()
    val k=sc.nextInt()
    val arr=IntArray(n){0}
    for(i in 0 until n)
        arr[i]=sc.nextInt()
    var answer=Int.MIN_VALUE
    var sum=0
    for(i in 0 ..n-k){
        for(j in i until i+k)
            sum+=arr[j]
        answer= max(answer,sum)
        sum=0
    }
    println(answer)
}